### DISCUSSION TITLE
Request signInInitiate failed with message: self signed certificate in certificate chain - Windows Machine, Visual Studio, CoPilot

### DISCUSSION THREAD

**Discussion by mlsmith007** [community/community#41866] [Copilot Conversations]
### Select Topic Area

Bug

### Body

I work for a Corporation that uses a Proxy that does SSL inspection. I use Visual Studio Professional on a Windows Box and am having the "Request signInInitiate failed with message: self signed certificate in certificate chain" issue dues to this. Excluding the URLs from SSL inspection is not a solution, due to Corporate Policies. At 70,000+ employees the corporation is not very likely to change its policies on SSL exceptions to work around what is essentially a Bug in a product add-on.

I'm making this post, to one track possible solutions to this issue and two, to express the importance of this issue, since we (Corporations) are a large chunk of your potential customers and any reasonably large Corporation will have similar security policies in place. 

CoPilot has the potential to greatly increase developers' productivity which presents value to corporations.

I sincerely hope this get's the attention it deserves and is fixed as soon as possible.

- Mike

**Comment by siddharthgoel88** [comment]
> 

I found a solution for this which works for me. I have blogged about it at  https://sidd.io/2023/01/github-copilot-self-signed-cert-issue/ . I tried with Intellij but high level solution should be same for Visual Studio as well.

Fix is as follows :

1. Export the self-signed certificate in discussion
2. Convert it into `.pem` format if not already
3. Export the path of this `.pem` cert to `NODE_EXTRA_CA_CERTS` variable
4. Restart your Visual Studio and it should work

**Comment by tilmann1** [comment]
@mlsmith007 Copilot recently added support for self-signed certificates for Copilot for Business users, please see https://docs.github.com/en/enterprise-cloud@latest/copilot/troubleshooting-github-copilot/troubleshooting-certificate-errors-for-github-copilot for more details. Thanks!

**Reply by CristianTGT** [reply]
Is there a way to find out what certificate is causing the issue?
